### PR TITLE
BOAC-1524: builds curated group student list in Vue

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1524

Some TODOs left in there; will tackle those in a subsequent PR.

### CODE REVIEW

File: src/components/curated/CuratedGroupList.vue [hunk 1]
```diff
@@ -20,11 +20,23 @@
       <strong>{{ curatedGroup.name }}</strong> group:
       <SearchStudents :withButton="true"/>
     </div>
+    <div v-if="curatedGroup.students.length">
+      <div id="curated-cohort-students" class="list-group">
+        <CuratedGroupStudent :student="student"
+                             :sort="sort"
+                             class="list-group-item student-list-item"
+                             :class="anchor === student.uid ? 'list-group-item-info' : ''"
```
Comment (johncrossman) [inline src/components/curated/CuratedGroupList.vue]: @lyttam Equivalent to the line above is `:class="{'list-group-item-info': anchor === student.uid}"` which seems to be our convention. Your call.

File: src/components/curated/CuratedGroupList.vue [hunk 2]
```diff
@@ -53,10 +66,60 @@ export default {
...
-    this.orderBy.options = _.filter(options, 'available');
+    this.sort.options = _.filter(options, 'available');
+  },
+  computed: {
+    orderedStudents() {
+      return _.orderBy(this.curatedGroup.students, this.studentComparator);
+    },
+    anchor() {
+      return location.hash;
+    }
```
Comment (johncrossman) [inline src/components/curated/CuratedGroupList.vue]: @lyttam Very much _not_ a merge blocker but the functions above could be one-liners. Eg, `anchor() => location.hash`

File: src/components/curated/CuratedGroupHeader.vue [hunk 3]
```diff
@@ -24,8 +24,9 @@
         </form>
       </div>
       <div class="has-error"
-           data-ng-bind="renameMode.error"
-           data-ng-if="renameMode.error && !renameMode.hideError"></div>
+           v-if="renameMode.error && !renameMode.hideError">
+        {{ renameMode.error }}
+      </div>
       <div class="faint-text">255 character limit <span data-ng-if="renameMode.input.length">({{255 - renameMode.input.length}} left)</span></div>
```
Comment (johncrossman) [inline src/components/curated/CuratedGroupHeader.vue]: @lyttam We want `v-if` above?

File: src/components/curated/CuratedGroupStudent.vue [hunk 4]
```diff
@@ -1,13 +1,158 @@
...
+      <button :id="`student-${student.uid}-curated-cohort-remove`"
+              class="btn btn-link"
+              @click="removeFromCuratedCohort">
+        <i class="fas fa-times-circle"></i>
+      </button>
+    </div>
+    <div class="cohort-list-view-column-01">
+      <StudentAvatar :size="'large'"
+                     :student="student"
+                     :alertCount="student.alertCount"/>
```
Comment (johncrossman) [inline src/components/curated/CuratedGroupStudent.vue]: @lyttam Nice. We have a Jira to do the `alertCount` work (as above): https://jira.ets.berkeley.edu/jira/browse/BOAC-1534  Assign to yourself it work is done.

File: src/components/curated/CuratedGroupStudent.vue [hunk 5]
```diff
@@ -1,13 +1,158 @@
...
+      <div class="cohort-student-name-container">
+        <div>
+          <a :id="student.uid" :href="`/student/${student.uid}`">
+            <h3 class="student-name"
+                :class="user.inDemoMode ? 'demo-mode-blur' : ''"
+                v-if="sort.selected !== 'first_name'">
+              {{ student.lastName }}, {{ student.firstName }}
+            </h3>
+            <h3 class="student-name"
+                :class="user.inDemoMode ? 'demo-mode-blur' : ''"
```
Comment (johncrossman) [inline src/components/curated/CuratedGroupStudent.vue]: Ditto on `:class` convention.

**Comment by johncrossman** [comment]
Good work, @lyttam  Amend as you see fit. I'll merge when ready.

**Comment by lyttam** [comment]
@johncrossman Thanks, I incorporated your feedback.

**Comment by johncrossman** [comment]
Thanks @lyttam  Waiting on T

**Pull request merged**